spec/controllers/queue_items_controller_spec.rb

```diff
@@ -155,45 +154,278 @@
+                                         "#{qi_2.id}" => { position: 3, user_rating: 2 },
+                                         "#{qi_3.id}" => { position: 2, user_rating: 2 }
+                                       }
+          end
+
+          it "normalizes position numbers" do                              
+            expect(qi_1.reload.position).to eq(3)
+            expect(qi_2.reload.position).to eq(2)
+          end
+
+          it "sets an info message" do
+            expect(flash[:info]).to be_present
+          end
+
+          it "redirects back" do
+            expect(response).to redirect_to queue_path
+          end
+        end
+
+        context "when two positions are set to the same number" do
+          let(:qi_1) { Fabricate(:queue_item, position: 1, user: pete) }
+          let(:qi_2) { Fabricate(:queue_item, position: 2, user: pete) }
+
+          before do 
+            post :update, queue_items: { "#{qi_1.id}" => { position: 1, user_rating: 2 }, 
+                                         "#{qi_2.id}" => { position: 1, user_rating: 2 } 
+                                        }
+          end
+
+          it "doesn't change any position numbers" do
+            expect(qi_1.position).to eq(1)
+            expect(qi_2.position).to eq(2)
+          end
+
+          it "sets a danger message" do
+            expect(flash[:danger]).to be_present
+          end
+
+          it "redirects back" do
+            expect(response).to redirect_to queue_path
+          end
+        end
+
+        context "when validations fail" do 
+          context "when a position is set to a float" do
+            let(:qi_1) { Fabricate(:queue_item, position: 1, user: pete) }
+            let(:qi_2) { Fabricate(:queue_item, position: 2, user: pete) }
+
+            before do
+              post :update, queue_items: { "#{qi_1.id}" => { position: 1, user_rating: 2 }, 
+                                           "#{qi_2.id}" => { position: 1.5, user_rating: 2 } 
+                                          }
+            end
+
+            it "does not update any queue items" do
+              expect(qi_1.reload.position).to eq(1)
+              expect(qi_2.reload.position).to eq(2)
+            end
+
+            it "sets a danger message" do
+              expect(flash[:danger]).to be_present
+            end
+
+            it "redirects back" do
+              expect(response).to redirect_to queue_path
+            end
+          end
+
+          context "when a position contains a non-digit character" do
+            let(:qi_1) { Fabricate(:queue_item, position: 1, user: pete) }
+            let(:qi_2) { Fabricate(:queue_item, position: 2, user: pete) }
+
+            before do
+              post :update, queue_items: { "#{qi_1.id}" => { position: "1123a", user_rating: 2 }, 
+                                           "#{qi_2.id}" => { position: 1.5, user_rating: 2 } 
```

Comment:
This is spilling over from your previous context. This could fail based on the float as well. I would recommend that you have "1123a" and "1" for this test scenario.